### PR TITLE
Add platform independent sample script for headless init

### PR DESCRIPTION
**Pull request by sytolk**
_Issue #, if available:_ https://github.com/aws-amplify/amplify-cli/issues/9065

_Description of changes:_

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by josefaidt — CHANGES_REQUESTED** [review]
Hey @sytolk thanks for raising this! While I think it is a great idea to write a platform-agnostic headless CLI script to initialize a project, I feel there is missing context here regarding `amplify.json` as the contents appear to be `amplify/.config/project-info.json`. Is this a sample to initialize an existing project or programmatically initialize a new project?

**Comment by sytolk** [comment]
@josefaidt this script is the same like this: 
https://github.com/aws-amplify/docs/blob/main/src/pages/cli/usage/headless.mdx#headless-category-payloads
but for the init there is only bash script now: 
https://github.com/aws-amplify/docs/blob/main/src/pages/cli/usage/headless.mdx#sample-script
where is `amplify/.config/project-info.json` in this bash script now ?? amplify.json is the same like in bash script.. simple I not understand you question :)

**Comment by josefaidt** [comment]
Hey @sytolk :wave: apologies for the delay here!!! As mentioned I do think the platform-agnostic script is a great addition, however I would prefer not introducing a third party dependency (`shelljs`), and is it possible to include the JSON blob as a string in the script rather than having a separate file? If not, can you rename the `amplify.json` sample name something like `amplify-headless-init-payload.json`?

**Comment by sytolk** [comment]
Hi, @josefaidt I have renamed JSON file to amplify-headless-init-payload.json. In this doc `shelljs` is here only to be clear for developers that it's need to execute this command line. They will choose the command line execution library `execa` `shelljs` or simple node `child_process`. But if this third party dependency bothers you feels free to replace it.
_is it possible to include the JSON blob as a string_ - yes it's possible but I think this will be very bad design (this can be done with webpack build and.. simple let the developers choose this too). This is documentation only and needs to have only samples here :)

File: src/pages/cli/usage/headless.mdx [hunk 1]
```diff
@@ -177,6 +177,53 @@ Installs, initializes, and provisions resources for a sample amplify application
...
+}
+},
+"providers": {
+"awscloudformation": {
+"configLevel": "project",
+"useProfile": false,
+"profileName": "default"
+}
+}
+}
```
Comment (josefaidt) [inline src/pages/cli/usage/headless.mdx:223]:
```suggestion
{
  "amplify": {
    "projectName": "headlessProjectName",
    "envName": "myenvname",
    "defaultEditor": "code"
  },
  "frontend": {
    "frontend": "javascript",
    "framework": "react",
    "config": {
      "SourceDir": "src",
      "DistributionDir": "build",
      "BuildCommand": "npm run-script build",
      "StartCommand": "npm run-script start"
    }
  },
  "providers": {
    "awscloudformation": {
      "configLevel": "project",
      "useProfile": false,
      "profileName": "default"
    }
  }
}

```

File: src/pages/cli/usage/headless.mdx [hunk 2]
```diff
@@ -177,6 +177,53 @@ Installs, initializes, and provisions resources for a sample amplify application
...
+JSON.stringify(amplify.amplify).replaceAll('"', '\\"') +
+'" --frontend "' +
+JSON.stringify(amplify.frontend).replaceAll('"', '\\"') +
+'" --providers "' +
+JSON.stringify(amplify.providers).replaceAll('"', '\\"') +
+'" --yes';
+if (shell.exec(cmd).code !== 0) {
+shell.echo('Error amplify init');
+shell.exit(1);
+}
```
Comment (josefaidt) [inline src/pages/cli/usage/headless.mdx:195]:
```suggestion
const shell = require('shelljs');
const amplify = require('./amplify-headless-init-payload.json');
const cmd =
  'amplify init --amplify "' +
  JSON.stringify(amplify.amplify).replaceAll('"', '\\"') +
  '" --frontend "' +
  JSON.stringify(amplify.frontend).replaceAll('"', '\\"') +
  '" --providers "' +
  JSON.stringify(amplify.providers).replaceAll('"', '\\"') +
  '" --yes';
  
if (shell.exec(cmd).code !== 0) {
  shell.echo('Error amplify init');
  shell.exit(1);
}
```

**Comment by josefaidt** [comment]
Hey @sytolk thanks for the clarification and for the adjustment! I've left a small formatting suggestion and I think we'll be good to go!

**Review by katiegoines — DISMISSED** [review]
lgtm!

**Review by josefaidt — APPROVED** [review]
LGTM, thank you @sytolk !! 🙏

**Review by kuhlmanp — APPROVED** [review]

**Review by katiegoines — APPROVED** [review]

**Review by kuhlmanp — APPROVED** [review]

**Review by josefaidt — APPROVED** [review]

**Review by josefaidt — DISMISSED** [review]

**Review by kuhlmanp — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Review by josefaidt — APPROVED** [review]

**Pull request merged**